sdk/eventhub/azure-eventhub/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,228 @@
+* [Ensure your SAS token is generated correctly][AuthorizeSAS]
+
+[Troubleshoot authentication and authorization issues with Event Hubs][troubleshoot_authentication_authorization] lists other possible solutions.
+
+## Connectivity issues
+
+### Timeout when connecting to service
+
+* Verify that the connection string or fully qualified domain name specified when creating the client is correct.  [Get an Event Hubs connection string][GetConnectionString] demonstrates how to acquire a connection string.
+* Check the firewall and port permissions in your hosting environment and that the AMQP ports 5671 and 5762 are open.
+  * Make sure that the endpoint is allowed through the firewall.
+* Try using WebSockets, which connects on port 443.  See [configure web sockets][PublishEventsWithWebSocketsAndProxy] sample.
+* See if your network is blocking specific IP addresses.
+  * [What IP addresses do I need to allow?][EventHubsIPAddresses]
+* If applicable, check the proxy configuration.  See [configure proxy][PublishEventsWithWebSocketsAndProxy] sample.
+* For more information about troubleshooting network connectivity is at [Event Hubs troubleshooting][EventHubsTroubleshooting]
+
+### SSL handshake failures
+
+This error can occur when an intercepting proxy is used.  We recommend testing in your hosting environment with the proxy disabled to verify.
+
+### Socket exhaustion errors
+
+Applications should prefer treating the Event Hubs clients as a singleton, creating and using a single instance through the lifetime of their application.  This is important as each client type manages its connection; creating a new Event Hub client results in a new AMQP connection, which uses a socket.  Additionally, it is essential to be aware that your client is responsible for calling `close()` when it is finished using a client or to use the `with statement` for clients so that they are automatically closed after the flow execution leaves that block.
+
+
+### Connect using an IoT connection string
+
+Because translating a connection string requires querying the IoT Hub service, the Event Hubs client library cannot use it directly.  The [IoT Hub Connection String Sample][IoTConnectionString] sample describes how to query IoT Hub to translate an IoT connection string into one that can be used with Event Hubs.
+
+Further reading:
+* [Control access to IoT Hub using Shared Access Signatures][IoTHubSAS]
+* [Read device-to-cloud messages from the built-in endpoint][IoTEventHubEndpoint]
+
+### Cannot add components to the connection string
+
+The legacy Event Hub clients allowed customers to add components to the connection string retrieved from the portal.  The legacy clients are in packages [com.microsoft.azure:azure-eventhubs][MavenAzureEventHubs] and [com.microsoft.azure:azure-eventhubs-eph][MavenAzureEventHubsEPH].  The current generation supports connection strings only in the form published by the Azure portal.
+
+#### Adding "TransportType=AmqpWebSockets"
```

Comment:
The text in quotes is significant here, as that is the exact slug that had previously worked for connection strings.    To ensure that it can be found via search, we'll want to leave that text.    I thought the heading read a bit awkwardly and went with `"TransportType=AmqpWebSockets" Alternative` in the .NET guide.